File: backend/src/main/java/com/ssafy/cheket/service/ticket/TicketService.java
package com.ssafy.cheket.service.ticket;

import com.ssafy.cheket.dto.ticket.response.GetUpcomingTicketResponse;
import com.ssafy.cheket.dto.ticket.response.GetUsedAndExpiredTicketResponse;

import java.util.List;

public interface TicketService {

    // 티켓 구매
    Long purchaseTickets(Long userId, Long showId, Long sessionId, String seatAccessToken, List<Long> sessionSeatIds);

    // 티켓 환불
    void refundTicket(Long ticketId);

    // 볼 예정인 티켓 조회
    List<GetUpcomingTicketResponse> getUpcomingTickets(Long id);

    // 관람 완료 / 만료 티켓 조회
    List<GetUsedAndExpiredTicketResponse> getUsedAndExpiredTickets(Long id);

    // 티켓 양도
    Long transferTicket(Long senderUserId, Long ticketId, String receiverPhoneNumber);

}

File: frontend/apps/customer-app/android/app/src/main/java/com/ssafy/cheket/core/network/service/TicketService.kt
package com.ssafy.cheket.core.network.service

import com.ssafy.cheket.core.network.dto.*
import retrofit2.http.*

interface TicketService {

    @POST("api/v1/shows/{showId}/sessions/{sessionId}/purchase")
    suspend fun purchaseTicket(
        @Header("Seat-Access-Token") seatAccessToken: String,
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
        @Body request: PurchaseRequest,
    ): ApiResponse<PurchaseResponse>

    @POST("api/v1/shows/{showId}/sessions/{sessionId}/seats")
    suspend fun lockSeats(
        @Path("showId") showId: Long,
        @Path("sessionId") sessionId: Long,
        @Body request: SeatLockRequest,
    ): ApiResponse<Unit>

    @GET("api/v1/tickets/upcoming")
    suspend fun getUpcomingTickets(): ApiResponse<List<UpcomingTicketDto>>

    @GET("api/v1/tickets/collection")
    suspend fun getCollectionTickets(): ApiResponse<List<CollectionTicketDto>>

    @POST("api/v1/tickets/{ticketId}/transfer")
    suspend fun transferTicket(
        @Path("ticketId") ticketId: Long,
        @Body request: TransferRequest,
    ): ApiResponse<Unit>

    @POST("api/v1/tickets/{ticketId}/qr")
    suspend fun generateQr(@Path("ticketId") ticketId: Long): ApiResponse<QrResponse>

    @POST("api/v1/tickets/{ticketId}/refund")
    suspend fun refundTicket(@Path("ticketId") ticketId: Long): ApiResponse<Unit>

}
